cypress/integration/settings/username-change.js

```diff
@@ -0,0 +1,180 @@
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('mruggesh', { force: true });
+
+    cy.contains(
+      'Please note, changing your username will also change ' +
+        'the URL to your profile and your certifications.'
+    )
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-info');
+  });
+
+  // eslint-disable-next-line
+  it('Should be able to click the `Save` button if username is avalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('ojeyton', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('not.be.disabled');
+    });
+  });
+
+  it('Should show username is unavalable if it is unavalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('twaha', { force: true });
+
+    cy.contains('Username not available.')
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-warning');
+  });
+
+  // eslint-disable-next-line
+  it('Should not be able to click the `Save` button if username is unavalable', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('twaha', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('be.disabled');
+    });
+  });
+
+  it('Should not show anything if user types their current name', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('developmentuser', { force: true });
+  });
+
+  // eslint-disable-next-line
+  it('Should not be able to click the `Save` button if user types their current name', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('developmentuser', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('be.disabled');
+    });
+  });
+
+  it('Should show warning if username includes invalid character', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('Quincy Larson', { force: true });
+
+    cy.contains('Username contains invalid characters.')
+      .should('be.visible')
+      .should('have.attr', 'role', 'alert')
+      .should('have.class', 'alert alert-danger');
+  });
+
+  // eslint-disable-next-line
+  it('Should not be able to click the `Save` button if username includes invalid character', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('Quincy Larson', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.contains('Save').should('be.disabled');
+    });
+  });
+
+  it('Should change username if `Save` button is clicked', () => {
+    cy.get('@usernameInput')
+      .clear({ force: true })
+      .type('quincy', { force: true });
+
+    cy.get('@usernameForm').within(() => {
+      cy.wait(2000);
+      cy.contains('Save').click({ force: true });
+    });
```

Comment:
```suggestion
      .type('quincy', { force: true });

    cy.contains('Username is available.');

    cy.get('@usernameForm')
      .contains('Save')
      .should('be.disabled');
```
Rather than an arbitrary wait, we can use `contains` to wait for the condition we're interested in (the server has reported back)